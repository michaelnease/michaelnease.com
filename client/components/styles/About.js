import styled from "styled-components";

export const AboutStyled = styled.section`
  height: 80rem;
  position: relative;
  background: url("/static/images/bio/bio.gif") no-repeat center center;
  background-size: cover;
  border-bottom: 10px solid ${props => props.theme.colors.black};
  @media (max-width: 700px) {
    height: 100rem;
  }
`;

export const Text = styled.div`
  width: 60rem;
  p {
    text-align: justify;
    font-weight: 300;
    font-size: 1.7rem;
    margin-bottom: 20px;
  }
  @media (max-width: 700px) {
    padding: 3rem;
  }
`;

export const Rex = styled.img`
  box-shadow: ${props => props.theme.effects.bs};
  border: 1px solid ${props => props.theme.effects.black};
  float: left;
  width: 20rem;
  margin: ${props => props.theme.spacing.margin};
  position: relative;
  top: 10px;
`;
