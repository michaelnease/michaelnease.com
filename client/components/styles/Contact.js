import styled from "styled-components";

export const Form = styled.form`
  display: flex;
  position: relative;
  align-content: center;
  justify-content: center;
  background-color: ${props => props.theme.colors.darkGray};
  border-bottom: 10px solid ${props => props.theme.colors.black};
  padding: 3rem;
`;

export const Inputs = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 121.5rem;
  grid-gap: 1.5rem;

  input,
  textarea {
    border: none;
    background-color: ${props => props.theme.colors.gray};
    padding-left: 3rem;
    font-size: 1.5rem;
    box-shadow: ${props => props.theme.effects.bs};
  }

  input {
    height: 7.4rem;
  }

  textarea {
    padding: 3rem;
    grid-column: span 2;
  }

  @media (max-width: 700px) {
    input,
    textarea {
      grid-column: span 2;
    }
  }
`;

//7.4 + 1.5 + 7.4 + 1.5
