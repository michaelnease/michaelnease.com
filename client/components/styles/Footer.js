import styled from "styled-components";

export const Foot = styled.footer`
  padding: 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 10rem;
  p {
    color: ${props => props.theme.colors.gray};
  }

  @media (max-width: 700px) {
    padding: 1rem;
    margin: 0;
  }
`;
