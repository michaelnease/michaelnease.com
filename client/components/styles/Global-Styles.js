import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  @font-face {
    font-family: 'Roboto';
    src: url('/static/fonts/Roboto-Regular.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
    font-family: 'Roboto';
    src: url('/static/fonts/Roboto-Light.ttf') format('truetype');
    font-weight: 300;
    font-style: normal;
  }

  html {
    box-sizing: border-box;
    font-size: 10px;
  }

  *, *:before, *:after {
    box-sizing: inherit;
  }

  body, h1, h2, h3, h4, h5, h6, p, ol, ul {
    margin: 0;
    padding: 0;
    font-weight: normal;
  }

  ol, ul {
    list-style: none;
  }

  img {
    max-width: 100%;
    height: auto;
  }

  body {
    padding: 0;
    margin: 0;
    font-size: 1.5rem;
    line-height: 2;
    font-family: "Roboto", Arial, Helvetica, sans-serif;
  }

  a {
    text-decoration: none;
  }
  
  button { font-family: "Roboto", Arial, Helvetica, sans-serif; }

  button:focus {
    outline:0;
  }

  textarea:focus, input:focus{
    outline: none;
  }
`;

export default GlobalStyles;
