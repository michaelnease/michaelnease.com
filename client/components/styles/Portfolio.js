import styled from "styled-components";
import Image from "../Image";

export const PortfolioStyled = styled.section`
  position: relative;
  border-bottom: 10px solid ${props => props.theme.colors.black};
`;

export const Work = styled.section`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 20px;
  width: 80rem;
  margin: 10rem 0;
  @media (max-width: 700px) {
    padding: 0 1rem;
  }
`;

export const PortfolioImage = styled(Image)`
  grid-column: span ${props => props.span};
  border: 2px solid ${props => props.theme.colors.black};
  box-shadow: ${props => props.theme.effects.bs};
`;
