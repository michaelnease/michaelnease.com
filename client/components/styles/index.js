export * from "./Header";
export * from "./Quotes";
export * from "./About";
export * from "./Technology";
export * from "./Portfolio";
export * from "./Contact";
export * from "./Footer";
export * from "./Send-Message-Button";

export * from "./Texture";
