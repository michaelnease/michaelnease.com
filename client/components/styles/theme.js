export const theme = {
  colors: {
    black: "#141414",
    white: "#fff",
    gray: "#e0e0e0",
    darkGray: "#a9a9a9"
  },
  spacing: {
    margin: "15px",
    padding: "15px"
  },
  effects: {
    bs: "-5px 5px 5px 0px rgba(0, 0, 0, 0.2)"
  },
  bar: {
    height: "60px"
  }
};
