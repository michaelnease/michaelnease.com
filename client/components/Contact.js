import { useReducer } from "react";
import SendMessageButton from "./Send-Message-Button";
import { Form, Inputs, Link } from "./styles";

const Contact = () => {
  const initialState = {
    name: "",
    email: "",
    message: "",
    data: {},
    isNameValid: false,
    isEmailValid: false,
    isMessageValid: false,
    isSendingMessage: false
  };

  const reducer = (state, action) => {
    const {
      name,
      email,
      message,
      data,
      isNameValid,
      isEmailValid,
      isMessageValid,
      isSendingMessage
    } = action;
    switch (action.type) {
      case "name":
        return { ...state, name, isNameValid };
      case "email":
        return { ...state, email, isEmailValid };
      case "message":
        return { ...state, message, isMessageValid };
      case "data":
        return { ...state, data };
      case "isSendingMessage":
        return { ...state, isSendingMessage };
      case "clearForm":
        return {
          ...state,
          name: "",
          email: "",
          message: "",
          isEmailValid: false,
          isMessageValid: false,
          isNameValid: false
        };
    }
  };

  const [state, dispatch] = useReducer(reducer, initialState);

  const handleNameInput = name => {
    const isNameValid = /^[a-zA-Z ]{2,45}$/.test(name);
    dispatch({ type: "name", name, isNameValid });
  };

  const handleEmailInput = email => {
    const isEmailValid = /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
    dispatch({ type: "email", email, isEmailValid });
  };

  const handleMessageInput = message => {
    const isMessageValid = message.length > 5 ? true : false;
    dispatch({ type: "message", message, isMessageValid });
  };

  const handleSendMessage = async () => {
    dispatch({ type: "isSendingMessage", isSendingMessage: true });
    const payload = {
      name,
      email,
      message
    };

    fetch("/contact", {
      method: "post",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(payload)
    })
      .then(response => response.json())
      .then(data => {
        dispatch({ type: "data", data });
        dispatch({ type: "isSendingMessage", isSendingMessage: false });
        dispatch({ type: "clearForm" });
      })
      .catch(error => {
        console.error(error);
        dispatch({ type: "clearForm" });
      });
  };

  const {
    name,
    email,
    message,
    isNameValid,
    isEmailValid,
    isMessageValid,
    isSendingMessage
  } = state;

  return (
    <Form>
      <Link name="contact" />
      <Inputs>
        <input
          type="text"
          maxLength="45"
          placeholder="Name"
          data-name="name"
          value={name}
          onChange={e => handleNameInput(e.target.value)}
        />
        <input
          type="email"
          maxLength="45"
          placeholder="Email"
          data-name="email"
          value={email}
          onChange={e => handleEmailInput(e.target.value)}
        />
        <textarea
          rows="8"
          cols="30"
          data-name="message"
          maxLength="400"
          placeholder="Message"
          value={message}
          onChange={e => handleMessageInput(e.target.value)}
        />
        <SendMessageButton
          text="Send Message"
          handleClick={handleSendMessage}
          disabled={
            !isNameValid || !isEmailValid || !isMessageValid || isSendingMessage
          }
          sending={isSendingMessage}
        />
      </Inputs>
    </Form>
  );
};

export default Contact;
