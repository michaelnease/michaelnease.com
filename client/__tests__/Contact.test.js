import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { render } from "react-testing-library";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";

import { theme } from "../components/styles/theme";
import Contact from "../components/Contact";

configure({ adapter: new Adapter() });

const updateInput = (wrapper, instance, value) => {
  const input = wrapper.find(instance);

  input.simulate("change", {
    target: { value: value }
  });

  return wrapper.find(instance);
};

describe("<Contact />", () => {
  const wrapper = shallow(<Contact />);

  it("Should allow users to fill out the <contact /> form", () => {
    const nameInput = updateInput(wrapper, '[data-testid="name"]', "Michael");
    const emailInput = updateInput(
      wrapper,
      '[data-testid="email"]',
      "michael@gmail.com"
    );
    const messageInput = updateInput(
      wrapper,
      '[data-testid="message"]',
      "Hello, nice to meet you"
    );

    expect(wrapper.find("SendMessageButton").prop("disabled")).toBe(false);
    expect(nameInput.props().value).toBe("Michael");
    expect(emailInput.props().value).toBe("michael@gmail.com");
    expect(messageInput.props().value).toBe("Hello, nice to meet you");
  });

  it("Should disable submit button if email is not valid", () => {
    const nameInput = updateInput(wrapper, '[data-testid="name"]', "Michael");
    const emailInput = updateInput(
      wrapper,
      '[data-testid="email"]',
      "michael@gmail"
    );
    const messageInput = updateInput(
      wrapper,
      '[data-testid="message"]',
      "Hello, nice to meet you"
    );

    expect(wrapper.find("SendMessageButton").prop("disabled")).toBe(true);
    expect(nameInput.props().value).toBe("Michael");
    expect(emailInput.props().value).toBe("michael@gmail");
    expect(messageInput.props().value).toBe("Hello, nice to meet you");
  });

  it("Should match Snapshot", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Contact />
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
