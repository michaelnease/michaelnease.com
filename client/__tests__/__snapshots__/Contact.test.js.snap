// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> Should match Snapshot 1`] = `
<DocumentFragment>
  .c1 {
  position: absolute;
  top: -60px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #a9a9a9;
  border-bottom: 10px solid #141414;
  padding: 3rem;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 80rem;
  grid-gap: 1.5rem;
}

.c2 input,
.c2 textarea {
  border: none;
  background-color: #e0e0e0;
  padding-left: 3rem;
  font-size: 1.5rem;
  box-shadow: -5px 5px 5px 0px rgba(0,0,0,0.2);
}

.c2 input {
  height: 5rem;
}

.c2 textarea {
  padding: 3rem;
  grid-column: span 2;
}

.c3 {
  font-size: 1.5rem;
  grid-column: 2;
  height: 5rem;
  border: none;
  background-color: #141414;
  color: #e0e0e0;
  box-shadow: -5px 5px 5px 0px rgba(0,0,0,0.2);
}

.c3:disabled,
.c3[disabled] {
  background-color: #e0e0e0;
  color: #a9a9a9;
}

.c3:active {
  background-color: #a9a9a9;
  color: #141414;
}

@media (max-width:700px) {
  .c2 input,
  .c2 textarea {
    grid-column: span 2;
  }
}

<form
    class="c0"
  >
    <a
      class="c1"
      name="contact"
    />
    <div
      class="c2"
    >
      <input
        data-testid="name"
        maxlength="45"
        placeholder="Name"
        type="text"
        value=""
      />
      <input
        data-testid="email"
        maxlength="45"
        placeholder="Email"
        type="email"
        value=""
      />
      <textarea
        cols="30"
        data-testid="message"
        maxlength="400"
        placeholder="Message"
        rows="8"
      />
      <div
        class="g-recaptcha"
        data-sitekey="6LdjqsgUAAAAAOYCCVDnZATGbPrgHcHSGKOjzmHe"
      />
      <button
        class="c3"
        data-testid="send"
        disabled=""
      >
        Send Message
      </button>
    </div>
  </form>
</DocumentFragment>
`;
